rsync_to_rdisc.py

```diff
@@ -73,36 +75,41 @@ def send_mail_incomplete(run, title_template, subject, run_file):
     send_email(subject, f"{title_template}.html", body_params)
 
 
-def check_rsync(run, folder, temperror, log):
-    if not Path(temperror).stat().st_size:
-        msg_bgarray_log = [[""], [">>> No errors detected <<<"]]
-        Path(temperror).unlink()  # remove tmperror file.
+def check_rsync(run, transfer_settings):
+    if not Path(settings.temp_error_path).stat().st_size:
+        log_msg = [[""], [">>> No errors detected <<<"]]
+        Path(settings.temp_error_path).unlink()  # remove tmperror file.
         rsync_result = "ok"
     else:
-        msg_bgarray_log = [
-            [""], [f">>>{run}_{folder} errors detected in Processed data transfer, not added to completed files <<<"]
+        log_msg = [
+            [""],
+            [f">>>{run}_{transfer_settings['name']} errors detected in data transfer, not added to completed files <<<"]
         ]
-        send_mail_transfer_state("{}{}".format(settings.folder_dic[folder]["input"], run), "error")
+        send_mail_transfer_state("{}{}".format(transfer_settings["input"], run), "error")
         rsync_result = "error"
-    with open(log, 'a', newline='\n') as log_file:
+
+    with open(settings.log_path, 'a', newline='\n') as log_file:
         log_file_writer = writer(log_file, delimiter='\t')
-        log_file_writer.writerows(msg_bgarray_log)
+        log_file_writer.writerows(log_msg)
+
     return rsync_result
 
 
 def check_daemon_running(wkdir):
+    run_file = Path(f"{wkdir}/transfer.running")
     try:
-        run_file = Path(f"{wkdir}/transfer.running")
         run_file.touch(exist_ok=False)
-        return run_file
     except FileExistsError:
         sys.exit()
+    else:
+        return run_file
```

Comment:
Maybe the `else` is not required, since the `except` has a `sys.exit()` and therefore would never continue. 
Agree that the `return` should be outside the `try` statement.

Reply:
So the else is executed if the try success. We can also do this:

```python
try:
    run_file.touch(exist_ok=False)
except FileExistsError:
    sys.exit()
return run_file
```